### PR TITLE
PTM-92 Use configured patient matching configuration

### PR DESCRIPTION
**Pull request by wluyima**
[PTM-92](https://issues.openmrs.org/browse/PTM-92) Use configured patient matching configuration

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -116,9 +122,11 @@
 	</dependencyManagement>
 
 	<properties>
-		<openMRSVersion>1.9.9</openMRSVersion>
+		<openMRSVersion>2.0.0</openMRSVersion>
 		<log4jScope>provided</log4jScope>
 		<commonsLangScope>provided</commonsLangScope>
+        <javaCompilerVersion>1.8</javaCompilerVersion>
+        <legacyUiVersion>0.1</legacyUiVersion>
```
Comment (dkayiwa) [inline pom.xml:129]: Is there any reason why you are not using the latest version of the legacyui module?

File: pom.xml [hunk 2]
```diff
@@ -116,9 +122,11 @@
 	</dependencyManagement>
 
 	<properties>
-		<openMRSVersion>1.9.9</openMRSVersion>
+		<openMRSVersion>2.0.0</openMRSVersion>
```
Comment (dkayiwa) [inline pom.xml:125]: Should this also be changed? https://github.com/openmrs/openmrs-module-patientmatching/blob/master/omod/src/main/resources/config.xml#L31

Comment (wluyima) [reply pom.xml:125]: Good catch! Added it

File: pom.xml (hunk 1 above)
Comment (wluyima) [reply pom.xml:129]: Based on our past practice, we always require the minimum supported version that way an implementation doesn't have to necessary have to run the latest legacyui version in order to use the latest version of patient matching with the new changes.

File: omod/src/main/resources/config.xml [hunk 3]
```diff
@@ -29,6 +29,9 @@
 	<!-- /Extensions -->
 
 	<require_version>1.7</require_version>
+    <require_modules>
```
Comment (dkayiwa) [inline omod/src/main/resources/config.xml:32]: Should we have both require and aware of the same module?

Comment (wluyima) [reply omod/src/main/resources/config.xml:32]: You got me confused earlier, this is why initially I had not added this, I think it should require and not be aware of

Comment (wluyima) [reply omod/src/main/resources/config.xml:32]: Updated

**Pull request merged**